harness/compareArray.js

```diff
@@ -26,8 +26,8 @@ compareArray.isSameValue = function(a, b) {
   return a === b;
 };
 
-compareArray.format = function(array) {
-  return `[${array.map(String).join(', ')}]`;
+compareArray.format = function(thingWithALengthPropertyAndNumericIndices) {
+  return `[${[].map.call(thingWithALengthPropertyAndNumericIndices, String).join(', ')}]`;
```

Comment:
@jugglinmike @leobalter how about now?

Reply:
Anything that doesn't suggest we expect `map` to exhaust an iterable.

Reply:
The original code was using map directly on the given array-like object. Now it only uses map on top of the same object. I'd rather advance any further modifications/improvements to this file into another PR, WDYT?

Reply:
Sure (I'm not requesting any further modifications)